File: gui/modules/buttons/buttons.rb
name 'BUTTONS'
hp 5
foreground [60, 100, 15]
background [60, 100, 97]

button_x = 5
negate_x = button_x + 6.5
port_x = negate_x + 7.7

top = 21
dy = 12.7

%w[A B C D E F G H].each_with_index do |name, i|
  button x: button_x, y: top + i * dy, label: name, label_size: :large
  button x: negate_x, y: top + i * dy, label: '¬', label_size: :large
  port x: port_x, y: top + i * dy
end

File: panelgen/internal/panel/buttons.go
package panel

import (
	"dhemery.com/DHE-Modules/panelgen/internal/control"
	"dhemery.com/DHE-Modules/panelgen/internal/svg"
)

func init() {
	registerBuilder("buttons", buttons)
}

func buttons() *Panel {
	const (
		hue = 60
		hp  = 5
	)

	var (
		fg          = svg.HslColor(hue, 1.0, 0.15)
		bg          = svg.HslColor(hue, 1.0, 0.97)
		button      = control.Button(bg, fg)
		negateLabel = labelAbove("¬", button, svg.LargeFont, fg)
		port        = control.Port(fg, bg)
	)

	p := NewPanel("BUTTONS", hp, fg, bg, "buttons")

	top := 21.0
	dy := 12.7
	button_x := 5.0
	negate_x := button_x + 6.5
	port_x := negate_x + 7.7

	labels := []string{"A", "B", "C", "D", "E", "F", "G", "H"}
	for i, name := range labels {
		y := top + float64(i)*dy

		p.Install(button_x, y, button)
		p.Install(negate_x, y, button)
		p.Install(port_x, y, port)

		p.Engrave(button_x, y, labelAbove(name, button, svg.LargeFont, fg))
		p.Engrave(negate_x, y, negateLabel)
	}

	return p
}
